### ISSUE TITLE
Post Group Call Does Not Work

### ISSUE THREAD

**Issue by turtle96** [turtle96/synchro-android#5]
code is in CreateGroupFragment, server call code is in SynchroAPI under postNewGroup method

**Comment by kfwong** [comment]
Working code. Please test before you commit :)

In CreateGroupFragment.java

``` java
@Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        View rootView = inflater.inflate(R.layout.fragment_create_group, container, false);
        return rootView;
    }

    @Override
    public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);


        final Spinner spinnerGroupType = (Spinner) this.getView().findViewById(R.id.spinnerGroupType);
        ArrayAdapter<String> adapterGroupType = new ArrayAdapter<String>(getContext(),
                android.R.layout.simple_spinner_dropdown_item, GROUP_TYPES_LIST);
        spinnerGroupType.setAdapter(adapterGroupType);

        final TextInputLayout layoutDesc = (TextInputLayout) this.getView().findViewById(R.id.input_layout_group_description);
        layoutDesc.setErrorEnabled(true);
        final EditText editTextDesc = (EditText) this.getView().findViewById(R.id.inputGroupDesc);

        editTextDesc.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {

            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                if (s.length() > DESC_MAX_LENGTH) {
                    layoutDesc.setError("Exceeded max character limit of 1000");
                }
                else {
                    layoutDesc.setError(null);
                }
            }

            @Override
            public void afterTextChanged(Editable s) {

            }
        });

        editTextDate = (EditText) this.getView().findViewById(R.id.inputGroupDate);
        editTextTime = (EditText) this.getView().findViewById(R.id.inputGroupTime);

        ImageButton buttonCalendar = (ImageButton) this.getView().findViewById(R.id.calendar_button);
        ImageButton buttonTime = (ImageButton) this.getView().findViewById(R.id.time_button);
        Button buttonCreate = (Button) this.getView().findViewById(R.id.create_group_button);

        buttonCalendar.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {showDatePickerDialog(view);
            }
        });
        buttonTime.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                showTimePickerDialog(view);
            }
        });

        final EditText editTextName = (EditText) this.getView().findViewById(R.id.inputGroupName);
        final EditText editTextVenue = (EditText) this.getView().findViewById(R.id.inputGroupVenue);

        buttonCreate.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {

                String groupName, groupType, groupDesc, groupDate, groupTime, groupVenue;
                groupName = editTextName.getText().toString().trim();
                groupType = spinnerGroupType.getSelectedItem().toString().trim();
                groupDesc = editTextDesc.getText().toString().trim();
                groupDate = editTextDate.getText().toString().trim();
                groupTime = editTextTime.getText().toString().trim();
                groupVenue = editTextVenue.getText().toString().trim();

                String[] fields = {groupName, groupType, groupDesc, groupDate, groupTime, groupVenue};

                //todo: will probably need some validation for groupname to ensure no duplicate names
                if (!checkFields(fields)) {
                    Snackbar checkFields = Snackbar.make(getView(), "Please make sure all fields are filled in :D",
                            Snackbar.LENGTH_INDEFINITE);
                    checkFields.setAction("ok", new View.OnClickListener() {
                        @Override
                        public void onClick(View v) {
                        }
                    });
                    checkFields.show();
                    return;
                }

                GroupData newGroupData = new GroupData(null, groupName, groupType, groupDesc, groupDate,
                        groupTime, time24Hour, groupVenue);

                SynchroAPI.getInstance().postNewGroup(newGroupData);

                Log.v("TADAH", "OMGOMG");

                //Toast.makeText(getContext(), "created group", Toast.LENGTH_LONG).show();

            /*
            FragmentManager manager = getFragmentManager();
            FragmentTransaction transaction = manager.beginTransaction().addToBackStack(null);
            transaction.setCustomAnimations(R.anim.fragment_fade_in, R.anim.fragment_fade_out,
                    R.anim.fragment_fade_in, R.anim.fragment_fade_out);

            ViewGroupFragment viewGroupFragment = ViewGroupFragment.newInstance();
            Bundle bundle = new Bundle();
            bundle.putSerializable(GET_GROUP_KEY, newGroupData);
            viewGroupFragment.setArguments(bundle);

            transaction.replace(R.id.content_fragment, viewGroupFragment);
            transaction.commit();
            */
            }
        });


    }
```

In SynchroAPI.java

``` java
 public void postNewGroup(GroupData group) {

        JsonObject groupJson = new JsonObject();
        groupJson.addProperty("name", group.getName());
        groupJson.addProperty("type", group.getType());
        groupJson.addProperty("description", group.getDescription());
        groupJson.addProperty("date_happening", group.getDate() + group.getTime24Hour());
        groupJson.addProperty("venue", group.getVenue());
        groupJson.addProperty("tags", group.getType() + " " + "test 123 fallala");


        System.out.println("HERE json 264 " + groupJson.toString());

        try {

            Ion.with(App.getContext())
                    .load("https://52.77.240.7/api/v1/groups")
                    .setHeader("Authorization", ivleAuthToken)
                    .setHeader("Content-Type", "application/json")
                    .setJsonObjectBody(groupJson)
                    .asJsonObject()
                    .setCallback(new FutureCallback<JsonObject>() {
                        @Override
                        public void onCompleted(Exception e, JsonObject result) {
                            //Toast.makeText(App.getContext(), "hello", Toast.LENGTH_LONG).show();
                            System.out.println("group " + result.toString());
                        }
                    });
/*
            JsonObject result;
            result = Ion.with(App.getContext())
                    .load(apiGroups)
                    .addHeader("Authorization", ivleAuthToken)
                    .asJsonObject()
                    .get();

            System.out.println("result " + result.toString());
*/

        } catch (Exception ex) {
            System.out.println("Error here" + ex.toString());
        }

    }
```

**Comment by turtle96** [comment]
Fixed by switching ion library version from 2.1.7 to 2.1.6
